.github/workflows/publish.yaml

```diff
@@ -0,0 +1,25 @@
+on:
+    push:
+        tags:
+            - v*
+
+jobs:
+    build:
+        name: Build & Publish
+        runs-on: ubuntu-latest
+        permissions:
+            contents: write
+        steps:
+            - name: Checkout
+              uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Nice convention 👍 I've always liked to do the same thing (pin the immutable hash + comment the tag).